metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -426,37 +506,49 @@ def loop_projections(
                         )
 
                         dataset_snapshot = DatasetSnapshot(
-                            urn=dataset_urn,
-                            aspects=[StatusClass(removed=False)],
+                            urn=dataset_urn, aspects=[StatusClass(removed=False)]
                         )
+
                         lineage_info = self._get_upstream_lineage_info_projection(
-                            dataset_urn, projection
+                            dataset_urn, inspector, projection, schema
                         )
 
                         if lineage_info is not None:
-                            yield MetadataChangeProposalWrapper(
-                                entityUrn=dataset_snapshot.urn, aspect=lineage_info
-                            ).as_workunit()
-                    except Exception as ex:
+                            upstream_lineage = lineage_info
+
+                            lineage_mcpw = MetadataChangeProposalWrapper(
+                                entityType="dataset",
+                                changeType=ChangeTypeClass.UPSERT,
+                                entityUrn=dataset_snapshot.urn,
+                                aspectName="upstreamLineage",
+                                aspect=upstream_lineage,
+                            )
+
+                            lineage_wu = MetadataWorkUnit(
+                                id=f"{self.platform}-{lineage_mcpw.entityUrn}-{lineage_mcpw.aspectName}",
+                                mcp=lineage_mcpw,
+                            )
+
+                            self.report.report_workunit(lineage_wu)
```

Comment:
why was this stuff changed - the as_workunit code was cleaner

Reply:
reverted back to as_workunit